Add local setup instructions for README

maikthomas opened the pull request:
#### What is this PR doing?
Adding local setup instructions to readme for running the app locally.

#### How should this be manually tested?
Formatted readme can be viewed on the branch: https://github.com/maikthomas/opentok-meet/tree/readme-local-setup
You could follow the steps I added to the readme and try to run to the app.

#### What are the relevant tickets?
There is no ticket for this.

v-kpheng commented on README.md:
````diff
@@ -2,25 +2,48 @@
+```
+
+- Run the server:
+
````
I think we're missing a step, `npm run postinstall`.

maikthomas replied:
Hmm yea I'm seeing this now. That should run after install but i'm getting a bunch of errors running `npm i` which probably stops it running.
Ideally we should rename that step to `build` or `webpack` and call it from `postinstall`, can we do that?

v-kpheng replied:
Hmm...that makes my spidey sense tingle, since it may break automated workflows.

We can introduce a new command instead, which combines commands and simplify workflows.  WDYT?

maikthomas replied:
I mean we should do:
```
scripts: {
"build": "webpack",
"postinstall": "npm run build",
...
}
```
this won't break anything because `postinstall` does the same thing as before and build doesn't exist yet.

maikthomas replied:
but happy to just add build, and not touch `postinstall` too.

v-kpheng replied:
@albertoacn or @msach22, can we rename `postinstall` to something easier to understand, like `build`?  Seems like being able to do so will make our lives simpler :-)

v-kpheng replied:
IOW, is there anything from preventing us from doing what we need to do in order to make things better?

maikthomas replied:
`postinstall` is an npm lifecycle script and can't be renamed as it is a reserved name that is invoked after npm install completes.
 We can however extract the build process to it's own command and call that in the postinstall command.

v-kpheng commented on package.json:
```diff
@@ -42,6 +42,7 @@
   },
   "scripts": {
     "postinstall": "webpack",
+    "build": "webpack",
```
How do you feel about making `build` an alias of `postinstall`?  This way, the intention is more clear?

maikthomas commented:
I think that's backwards, build should be the build process, if we want to
build on postinstall then the postinstall script should call the build
script, not the other way around

On Wed, 13 Jan 2021, 20:11 Kongpheng Nanthavongsa, <notifications@github.com>
wrote:

> *@v-kpheng* commented on this pull request.
> ------------------------------
>
> In package.json
> <https://github.com/opentok/opentok-meet/pull/92#discussion_r556765391>:
>
> > @@ -42,6 +42,7 @@
>    },
>    "scripts": {
>      "postinstall": "webpack",
> +    "build": "webpack",
>
> How do you feel about making build an alias of postinstall? This way, the
> intention is more clear?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentok/opentok-meet/pull/92#pullrequestreview-567537637>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD2V4ZPA5NWN6C3F2S7U4E3SZXV5FANCNFSM4WA72HYQ>
> .
>

v-kpheng commented:
> I think that's backwards, build should be the build process, if we want to build on postinstall then the postinstall script should call the build script, not the other way around

So you're a no-go for something like this?
```
"postinstall": "webpack",
"build": "npm run postinstall",
```

Or,
```
"postinstall": "webpack",
"build": "npm run install && npm run postinstall",
```

maikthomas commented:
Yes I think that's the wrong way around.
The build script should define how to build the app.
If we want to build the app after installing, then we can reference build
from the postinstall.
If we want to do some other stuff later in postinstall we can add that to
the postinstall script, and this shouldn't affect the build process.
If it was the other way around then changing what we want to do after
installing would affect what the build script does.


On Wed, 13 Jan 2021, 20:29 Kongpheng Nanthavongsa, <notifications@github.com>
wrote:

> I think that's backwards, build should be the build process, if we want to
> build on postinstall then the postinstall script should call the build
> script, not the other way around
>
> So you're a no-go for something like this?
>
> "postinstall": "webpack",
> "build": "npm run postinstall",
>
> Or,
>
> "postinstall": "webpack",
> "build": "npm run install && npm run postinstall",
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentok/opentok-meet/pull/92#issuecomment-759672534>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD2V4ZPRZPCXWBEWNM6FC5DSZXYDJANCNFSM4WA72HYQ>
> .
>

maikthomas commented:
postinstall is a script that npm runs after it installs packages, we can't
rename it, it's is a npm hook of sorts.


On Wed, 13 Jan 2021, 21:36 Kongpheng Nanthavongsa, <notifications@github.com>
wrote:

> *@v-kpheng* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/opentok/opentok-meet/pull/92#discussion_r556812610>:
>
> > +- [redis](https://redis.io/) (Recommended to install via homebrew on mac: [see instructions](https://medium.com/@petehouston/install-and-config-redis-on-mac-os-x-via-homebrew-eb8df9a4f298))
> +
> +## Running meet locally
> +
> +- Copy the contents of `config.json.sample` into `config.json` and add your credentials.
> +
> +- Ensure redis is running (e.g. `redis-server` on mac)
> +
> +- Install npm dependencies
> +
> +```
> +npm i
> +```
> +
> +- Run the server:
> +
>
> IOW, is there anything from preventing us from doing what we need to do in
> order to make things better?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentok/opentok-meet/pull/92#discussion_r556812610>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD2V4ZIV3QY5HC2L2HQFFOTSZX76DANCNFSM4WA72HYQ>
> .
>

maikthomas commented:
@albertoacn there's a line in the readme that says:
_If you wish to fork this project, please fork the parent project._
Referring to the aullman repo. 
It appears that this is no longer correct advice, the projects appear to have diverged and we are no longer keeping this up to date with aullman/opentok-meet.

If that's true then we should definitely remove this midleading info, wdyt?

albertoacn commented:
@maikthomas @v-kpheng 
IMHO I am more of a fan of:
```
scripts: {
"build": "webpack",
"postinstall": "npm run build",
...
}
```
I just find it more intuitive, i.e. as a post install action we build the app. Building the app here means webpack. So every action does what is expected.
I found
```
"postinstall": "webpack",
"build": "npm run postinstall", 
```
misleading since it feels we should run a post-install when building the app.

albertoacn commented:
> @albertoacn there's a line in the readme that says:
> _If you wish to fork this project, please fork the parent project._
> Referring to the aullman repo.
> It appears that this is no longer correct advice, the projects appear to have diverged and we are no longer keeping this up to date with aullman/opentok-meet.
> 
> If that's true then we should definitely remove this midleading info, wdyt?

yep, that's correct, this is the one ahead from a couple of years now. I would remove the fork parent repo line. People should just fork this repo (fork).

maikthomas commented:
@albertoacn I have updated the npm scripts to what we agreed.
I have also updated the disclaimer info, and I added extra info about the config to avoid the problem we saw about having mismatch between dev and prod config.
Please re-review when you can

albertoacn reviewed (approved):
LGTM
